projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts

```diff
@@ -250,18 +272,28 @@ export class IgxButtonGroupComponent implements AfterViewInit {
      *```
      */
     public selectButton(index: number) {
-        if (this.buttons.toArray()[index]._el.nativeElement.getAttribute('data-togglable') === 'false'
-            || this.buttons.toArray()[index]._el.nativeElement.classList.contains('igx-button--disabled')) {
+        const button = this.buttons[index];
+        const buttonElement = button.nativeElement;
+
+        if (buttonElement.classList.contains('igx-button--disabled')) {
             return;
         }
-        const buttonElement = this.buttons.toArray()[index]._el.nativeElement;
+
         this.selectedIndexes.push(index);
-        buttonElement.setAttribute('data-selected', true);
-        this.onSelect.emit({ button: this.buttons.toArray()[index], index });
-        this.values[index].selected = true;
+
+        this._renderer.setAttribute(buttonElement, 'data-selected', 'true');
```

Comment:
IMO we should just expose an input property on the button directive and use it instead of setting data- attributes.